src/resources/V2/Core/Events.ts

```diff
@@ -1,12 +1,58 @@
-// File generated from our OpenAPI spec
-
+// This file is manually maintained
 import {StripeResource} from '../../../StripeResource.js';
+
 const stripeMethod = StripeResource.method;
+
 export const Events = StripeResource.extend({
-  retrieve: stripeMethod({method: 'GET', fullPath: '/v2/core/events/{id}'}),
-  list: stripeMethod({
-    method: 'GET',
-    fullPath: '/v2/core/events',
-    methodType: 'list',
-  }),
+  constructor: function(...args: any) {
+    StripeResource.apply(this, args);
+  },
+  retrieve(...args: any[]) {
+    const transformResponseData = (response: any): any => {
+      return this.addFetchRelatedObjectIfNeeded(response);
+    };
+    return stripeMethod({
+      method: 'GET',
+      fullPath: '/v2/core/events/{id}',
+      transformResponseData,
+    }).apply(this, args);
+  },
+
+  list(...args: any[]) {
+    const transformResponseData = (response: any): any => {
+      return {
+        ...response,
+        data: response.data.map(this.addFetchRelatedObjectIfNeeded.bind(this)),
+      };
+    };
+    return stripeMethod({
+      method: 'GET',
+      fullPath: '/v2/core/events',
+      methodType: 'list',
+      transformResponseData,
+    }).apply(this, args);
+  },
+
+  addFetchRelatedObjectIfNeeded(pulledEvent: any) {
```

Comment:
Should we prefix the name of this function with `_` given that it will be visible to users when they inspect `stripe.v2.core.events`?

Reply:
v2 is a `V2Resource`, in which core is a `CoreResource`, in which events is an `EventsResource` which doesn't have this method.  just checked in vscode and all I see are list and retrieve.  but I could put a comment over it that says this is a stripe internal function and should not be called directly or something like that?

Reply:
I added a comment with a @private tag (I mimic'd another internal only function comment)

Reply:
Yes, Typescript helps us here coz we won't be including this method in the types. What I meant was a JS user can still inspect the object in hand (list all properties or view in debug mode) and find it. Rare, but doable